frontend/apps/filemanager/filemanagerbookinfo.lua

```diff
@@ -666,6 +679,103 @@ function BookInfo:editSummary(doc_settings_or_file, book_props)
     input_dialog:onShowKeyboard(true)
 end
 
+function BookInfo:showNotesFileDialog(notes_file, doc_settings_or_file, book_props)
+    local function saveNotesFile(new_notes_file)
+        if type(doc_settings_or_file) ~= "table" then -- no sidecar yet, create
+            doc_settings_or_file = DocSettings:open(doc_settings_or_file)
+        end
+        doc_settings_or_file:saveSetting("notes_file", new_notes_file)
+        if not self.is_current_doc then
+            if new_notes_file or doc_settings_or_file:readSetting("summary") then
+                doc_settings_or_file:flush()
+            else -- remove empty sidecar
+                local file = doc_settings_or_file:readSetting("doc_path")
+                doc_settings_or_file:purge(nil, { doc_settings = true }) -- keep custom
+                doc_settings_or_file = file -- to reopen bookinfo
+            end
+            self.summary_updated = true -- refresh FM
+        end
+        self.kvp_widget:onClose()
+        self:show(doc_settings_or_file, book_props)
+    end
+    local button_dialog, title, buttons
+    if notes_file then
+        title = notes_file
+        buttons = {
+            {
+                {
+                    text = _("Reset"),
+                    callback = function()
+                        UIManager:close(button_dialog)
+                        saveNotesFile()
+                    end,
+                },
+                {
+                    text = _("View"),
+                    callback = function()
+                        UIManager:close(button_dialog)
+                        TextViewer.openFile(notes_file)
+                    end,
+                },
+                {
+                    text = _("Edit"),
+                    enabled = self.ui.texteditor ~= nil,
+                    callback = function()
+                        UIManager:close(button_dialog)
+                        self.ui.texteditor:openFile(notes_file)
+                    end,
+                },
+            },
+        }
+    else
+        title = _("Notes file")
+        buttons = {
+            {
+                {
+                    text = _("Choose"),
+                    callback = function()
+                        UIManager:close(button_dialog)
+                        local PathChooser = require("ui/widget/pathchooser")
+                        local path_chooser = PathChooser:new{
+                            path = self.ui.file_chooser and self.ui.file_chooser.path or self.ui:getLastDirFile(),
```

Comment:
What would be the most confortable locations for such notes files ? Alongside books ? In koreader/settings/ ?
Dunno if you could/should store the last directory used for these, so next time one call "Choose", he gets in the directory where all his notes files are (instead of ui.getLastDirFile whatever that gives, or TextEditor last dir, dunno which is picked when).

Also wondering if there should be a default notes_file, a generic one, or the last one used! ? Having to go through "Choose" on each book may be bothering. I'm not a customer for this feature, but if I would, I'd probably use only a single file to append notes about all books I needed to write something about.